src/olympia/addons/tests/test_views.py

```diff
@@ -852,6 +852,47 @@ def test_too_many_categories(self):
         )
         assert response.status_code == 201, response.content
 
+    def test_set_slug(self):
+        response = self.client.post(
+            self.url,
+            data={
+                **self.minimal_data,
+                'slug': '!@!#!@##@$$%$#%#%$^^%&%',
+            },
+        )
+        assert response.status_code == 400, response.content
+        assert response.data == {
+            'slug': [
+                'Enter a valid “slug” consisting of letters, numbers, underscores or '
+                'hyphens.'
+            ]
+        }
+
+        DeniedSlug.objects.create(name='denied-slug')
+        response = self.client.post(
+            self.url,
+            data={
+                **self.minimal_data,
+                'slug': 'denied-slug',
+            },
+        )
+        assert response.status_code == 400, response.content
+        assert response.data == {
+            'slug': ['This slug cannot be used. Please choose another.']
+        }
+
+        response = self.client.post(
+            self.url,
+            data={
+                **self.minimal_data,
+                'slug': '1234',
```

Comment:
I forgot that `DeniedSlug` was the bit of code responsible for denying pure digits slugs and was initially struggling to figure out what caused that error message to be returned! Could you add a small comment ?

Reply:
https://github.com/mozilla/addons-server/pull/18299/commits/f8f35438a6ffef4e8e5184f8ee67d4d44294f064